samza-core/src/main/java/org/apache/samza/runtime/LocalApplicationRunner.java

```diff
@@ -200,6 +279,22 @@ static String createProcessorId(ApplicationConfig appConfig) {
     }
   }
 
+  private void cleanup() {
+    if(runIdLock != null) {
+      runIdLock.cleanState();
+    }
+    if(coordinationUtils != null) {
+      coordinationUtils.close();
+    }
+  }
+
+  private void stopProcessingAndShutDown() {
+    processors.forEach(StreamProcessor::stop);
+    cleanup();
+    appStatus = ApplicationStatus.UnsuccessfulFinish;
```

Comment:
qq. why is it Unsuccessful?

Reply:
stopProcessingAndShutDown is called when the global runid differs from this processor's runid or if the runid node itself is deleted. In the first case, it is a failure as this processor was using different intermediate streams than the rest of the processors and in the latter, there is no way for processors to agree upon the streams to use. THus, is it an unsuccessful run and needs to be indicated.

Reply:
I see why this is a confusion. can we take in an exception instead and set the state based on the exception?